### DISCUSSION TITLE
Metroidvania Con 2 Personajes

### DISCUSSION THREAD

**Discussion by LeandroCuello** [gamerevar/proyecto-comunitario-1-gd#50] [Archivados]
Premisa: Ambos juegos ocuririan en caminos paralelos que se unen al final. A lo largo del juego exiten obstaculos que se desbloquaran de forma "magica" o "casulidad" pero si uno juega ambos juegos se dara cuenta que esas casualidades ocurrian gracias al otro personaje.

Por ejemplo el personaje A no puede seguir su camino por culpa de un rio muy fuerte, Pero el personaje B justo tiene que tirar un tronco para pasar por ese mismo rio. Ahora el rio corre más lento y el personaje A puede pasar por el rio. Esto es solo un ejemplo, el juego podría ser futurista y hacer que A hackear unas puertas y justo eso ayude a B también.

Posible historia:
Dos tribus diferente mandan a sus elegidos a una misión por algo misterioso que se encuentra en una montaña un poder que puede salvar la aldea. Durante todo el viaje ambos elegidos se ayudan mutuamente pero sin darse cuenta creyendo que el camino se desbloquea por ayuda de los dioses. Al final del viaje se encuentran ambos en la entrada de la cueva y el jugador tiene que elegir al personaje para usar en la pelea final(una boss figth). Al entrar en la cueva hay un escrito antiguo que dice " el poder verdadero es el trabajo en equipo". (perdón no soy buen escritor)

Lo bueno:
- Los dos juegos pueden ser super distintos y lo unico que hay que cordinar son los "eventos" y que tena un boss final que sea el personaje de cada juego.
- Se puede contar la historias de dos personajes en paralelo y detalles del mundo.
- Los juegos a pesar de tener un final distinto (ya que en el de unity le tiene que ganar al de godot y viceversa) pueden tener un final similar. ej: el mal gana porque decidieron peliar en vez de unirse.
- Por lo que conozco es algo relativamente original.

Lo malo
-La historia y el word building tienen que tener coerencia en ambos juegos. Esto puede generar siertas limitaciones artisticas.

**Comment by roverem** [comment]
Es buena! La gente de GDQuest tiene un template de metroidvania https://github.com/GDquest/godot-platformer-2d

**Comment by LeandroCuello** [comment]
Recién lei la aclaración.Creoo que se puede sacar la mecánica de cambiar de personajes y quedaría un Metroidvania más clásico. Lo lindo es que cuando uno juegue al otro juego va a tener el giro argumental de que no eran espíritus y la boss figth del final va a ser más sentimental.

**Reply by gamerevar** [reply]
> 
> 
> Recién lei la aclaración.Creoo que se puede sacar la mecánica de cambiar de personajes y quedaría un Metroidvania más clásico. Lo lindo es que cuando uno juegue al otro juego va a tener el giro argumental de que no eran espíritus y la boss figth del final va a ser más sentimental.

Actualizalo en el primer post asi esta clara la idea. Esta buena.

**Reply by LeandroCuello** [reply]
ah genial no sabia que se podian editar los post!

**Comment by MasTho0128** [comment]
Este idea podria desarrollarse como un plataformero?

**Reply by LeandroCuello** [reply]
Ahora actualize la idea y en realidad podria ser lo que queramos.

**Comment by emrodk** [comment]
Me encanta la idea, se puede hacer que ambos juegos transcurran en distintos tiempos (futuro/pasado) o simplemente universos paralelos que dependan uno del otro para existir, el lugar final podría ser dónde ambos universos se solapan y si se basara en el tiempo dónde el futuro se une con el pasado

**Reply by LeandroCuello** [reply]
uyy me encanta! muy creativa la idea!

**Comment by doctorhibert** [comment]
Me gusta mucho la idea pero me parece que hacer un metroidvania sería un quilombo. Tal vez habría que hacer algo más simple, por lo menos para empezar.

Para agregar a tu idea, se me ocurrió que las cosas que vos decis como lo del tronco y eso sean obligatorias (así siempre va a pasar y no nos preocupamos por la continuidad entre los dos juegos).
Pero, al final del primer juego que jueges y consigas el ending donde matas al otro se arme un .txt como para decir que terminaste uno, y que cuando llegues al final del segundo juego se fije si tenés el .txt y te muestr el good ending donde el PC dice "Aguantá, ya viví esto, seamos amigos jaja salu2". (Podríamos bundlear los dos juegos en una misma carpeta en itch, y que haya unas subcarpetas tipo "The warrior's story y The mage's story)

**Reply by LeandroCuello** [reply]
Me encantó lo del los dos juegos estén en una carpeta 10/10! Los eventos tienen que ser obligatorios pero eso es tan fácil como decir "toca x para tirar el tronco". Hacer un Metroidvania no es tan complicado al menos que quieras que lo sea. Al menos eso creo. El mayor reto es diseñar un buen nivel.

**Reply by jh62** [reply]
Me gusta la idea de esta continuidad, pero así estarían aún más entrelazados los juegos y la comunicación entre los lead designers en este caso sería fundamental. 

En todo caso, si ambas historias pueden ocurrir en paralelo, sin necesariamente tener que experimentar eventos en común, se puede hacer que en cada juego haya si o si 3 eventos (por poner un número), pero que estos 3 eventos cruciales en cada uno puede ser cualquier cosa en cualquier momento. Lo único que harían los juegos entre si, es ver si evento 1 paso en juego 2, por ende tal cosa, si no paso, tal otra. Creo que permitiría diseñar ambos juegos de manera más individual.

**Reply by doctorhibert** [reply]
>En todo caso, si ambas historias pueden ocurrir en paralelo, sin necesariamente tener que experimentar eventos en común, se puede hacer que en cada juego haya si o si 3 eventos (por poner un número), pero que estos 3 eventos cruciales en cada uno puede ser cualquier cosa en cualquier momento. Lo único que harían los juegos entre si, es ver si evento 1 paso en juego 2, por ende tal cosa, si no paso, tal otra. Creo que permitiría diseñar ambos juegos de manera más individual.

Por eso decía, que en juego 1 sí o sí tenés que tirar un arbol, entonces no hace falta guardar ningun dato, porque en juego 2 ya va a estar el arbol caído. Si lo hacemos de esta manera no tenemos que enquilombarnos con mantener saves en dos juegos y coordinarlo entre los equipos; decimos al principio "bueno en el juego tiene que pasar esto esto y esto" y seguimos adelante.

**Comment by azagaya** [comment]
Creo que la idea se puede llevar a cabo, volviendo a la idea original de Leandro. En cada juego, algunas cosas pasan "por que si", y cuando jugas al otro, te das cuenta que el otro pj las había causado. Al pasar porque si, no hace falta que estén sincronizados. No hace falta que ambos juegos estén en la misma carpeta. Se pueden guardar los datos de los juegos en appdata en win, o en .local/share en linux, y cada juego puede buscar tmb el archivo de guardado del otro. Eso lo dejaría solo para desbloquear el final. El tema es que el final desbloqueado debería jugarse en uno de los dos juegos.

**Reply by LeandroCuello** [reply]
en realidad ambos juegos podrian tener el "final verdadero". por ejemplo en ambos juegos sin terminar el otro el personaje mata al boss y ahi esta el final malo. Pero si ya pasaste alguno de los juegos en vez de matar al boss lo dejas a poco de vida y le perdonas la vida. y el final verdadero puede ser una cutscene que es igual para ambos juegos.

**Reply by doctorhibert** [reply]
Claro, eso es lo que pensé yo, porque el bossfight final ya hay que hacerla, pero el good ending es simplemente diálogo.

**Reply by azagaya** [reply]
Como ambos juegos tendran estetica distinta, habria que de igual manera implementar el good ending en ambos. No?

**Comment by DeepGlowing** [comment]
Tu idea me ah dado una idea para desarrollar, en un rato abro otro hilo y la expongo, justo había pensado en lo del final conjunto luego de terminar las 2 juegos y veo que lo explicas en uno de los comentarios xD

**Comment by AlexandreInking** [comment]
La idea de emrodk me hizo recordar un anime que recomendaria ver para que se den una idea... Gunslinger Stratos: The Animation. Creo que está basada en un juego...
